kotlin/src/main/kotlin/io/logto/sdk/core/util/TokenUtils.kt

```diff
@@ -2,9 +2,36 @@ package io.logto.sdk.core.util
 
 import io.logto.sdk.core.extension.toIdTokenClaims
 import io.logto.sdk.core.type.IdTokenClaims
+import org.jose4j.jwa.AlgorithmConstraints
+import org.jose4j.jwk.JsonWebKeySet
+import org.jose4j.jws.AlgorithmIdentifiers
 import org.jose4j.jwt.consumer.JwtConsumerBuilder
+import org.jose4j.keys.resolvers.JwksVerificationKeyResolver
 
 object TokenUtils {
+    internal const val ISSUED_AT_RESTRICTIONS_IN_SECONDS = 60
+
+    fun verifyIdToken(
+        idToken: String,
+        clientId: String,
+        issuer: String,
+        jwks: JsonWebKeySet,
+    ) {
+        JwtConsumerBuilder().apply {
```

Comment:
did we apply "not expired" restriction here?

Reply:
i see the test case below. just curious does `JwtConsumerBuilder` not allow expired tokens by default?

Reply:
After inspecting [the source code](https://bitbucket.org/b_c/jose4j/src/ff6d6c2ee7b94ca5e1ee9c17b4647938110bd07d/src/main/java/org/jose4j/jwt/consumer/NumericDateValidator.java#lines-111:117), If we set the Jwt needs an expiration time, JwtConsumer will verify the expiration time.
So `JwtConsumer` does not allow expired tokens by default, but we can skip the validation by `        setSkipAllValidators()`

![image](https://user-images.githubusercontent.com/10806653/150139518-71a93a8a-1dd7-46e6-844d-a709d4f9f666.png)